lib/redis/distributed_store.rb

```diff
@@ -6,12 +6,12 @@ class DistributedStore < Distributed
     attr_reader :ring
 
     def initialize(addresses, options = { })
-      nodes = addresses.map do |address|
-        ::Redis::Store.new _merge_options(address, options)
-      end
-
       _extend_namespace options
-      @ring = Redis::HashRing.new nodes
+      @ring = options[:ring] || Redis::HashRing.new([], options[:replicas] || Redis::HashRing::POINTS_PER_SERVER)
+
+      addresses.each do |address|
+        @ring.add_node(::Redis::Store.new _merge_options(address, options))
```

Comment:
this feels like an unnecessary change to me, can you explain more about what it's doing?

Reply:
if you pass in a fully custom ring implementation, the nodes still need added to that ring. this is how "regular" Redis::Distributed does it as well: https://github.com/redis/redis-rb/blob/master/lib/redis/distributed.rb#L18

Reply:
cool, thanks!

Reply:
`HashRing` should do that for you: https://github.com/redis/redis-rb/blob/master/lib/redis/hash_ring.rb#L18-L20

Reply:
hah sorry i am an idiot...re-read what `HashRing#initialize` does and you definitely need this!